titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/KubeNotificationProcessor.java

```diff
@@ -85,6 +87,9 @@ public void enterActiveMode() {
                         return Mono.empty();
                     }
 
+                    if (event instanceof PodNotFoundEvent) {
+                        return handlePodNotFoundEvent((PodNotFoundEvent) event);
```

Comment:
as spoken in person, let's make sure this is not causing events to be potentially handled out of order (since we are `flatMapping` them all and processing them concurrently.

Reply:
I have added concurrency limit + a note (another PR).